inception-kb/src/main/java/de/tudarmstadt/ukp/inception/kb/KnowledgeBaseServiceImpl.java

```diff
@@ -1140,47 +1194,95 @@ else if (!stmts.hasNext()) {
+    private IndexWriter getIndexWriter(KnowledgeBase aKb) throws IOException
     {
         Analyzer analyzer = new StandardAnalyzer();
         Directory directory = FSDirectory
             .open(new File(luceneIndexDir, aKb.getRepositoryId()).toPath());
-        IndexWriter indexWriter = new IndexWriter(directory, new IndexWriterConfig(analyzer));
+        return new IndexWriter(directory, new IndexWriterConfig(analyzer));
+    }
 
-        try (RepositoryConnection conn = getConnection(aKb)) {
-            RepositoryResult<Statement> stmts = RdfUtils
-                .getStatementsSparql(conn, null, aKb.getLabelIri(), null,
-                    Integer.MAX_VALUE, false, null);
-            while (stmts.hasNext()) {
-                Statement stmt = stmts.next();
-                String id = stmt.getSubject().stringValue();
-                String label = stmt.getObject().stringValue();
-                String predicate = stmt.getPredicate().stringValue();
-                indexEntity(id, label, predicate, indexWriter);
+    @Override
+    public void indexLocalKb(KnowledgeBase aKb)
+    {
+        try {
+            IndexWriter indexWriter = getIndexWriter(aKb);
+
+            try (RepositoryConnection conn = getConnection(aKb)) {
+                RepositoryResult<Statement> stmts = RdfUtils
+                    .getStatementsSparql(conn, null, aKb.getLabelIri(), null, Integer.MAX_VALUE,
+                        false, null);
+                while (stmts.hasNext()) {
+                    Statement stmt = stmts.next();
+                    String id = stmt.getSubject().stringValue();
+                    String label = stmt.getObject().stringValue();
+                    indexCreatedConcept(id, label, indexWriter);
+                }
             }
+            indexWriter.close();
+        }
+        catch (IOException e) {
+            log.error("Could not index local KB.", e);
         }
-
-        indexWriter.close();
     }
 
-    private void indexEntity(String aId, String aLabel, String aPredictate,
-        IndexWriter aIndexWriter)
+    private void indexCreatedConcept(String aId, String aLabel, IndexWriter aIndexWriter)
     {
         try {
             String FIELD_ID = "id";
             String FIELD_CONTENT = "label";
+
             Document doc = new Document();
             doc.add(new StringField(FIELD_ID, aId, Field.Store.YES));
             doc.add(new StringField(FIELD_CONTENT, aLabel, Field.Store.YES));
+
             aIndexWriter.addDocument(doc);
             aIndexWriter.commit();
+            aIndexWriter.close();
+            log.info("LuceneIndex updated by creating Concept with id [{}] and label [{}].", aId,
+                aLabel);
+        }
+        catch (IOException e) {
+            log.error("Could not add concept with id [{}] and label [{}] to LuceneIndex.", aId,
+                aLabel);
+        }
+    }
+
+    private void indexUpdatedConcept(String aId, String aLabel, IndexWriter aIndexWriter)
```

Comment:
We shouldn't have to directly interact with the index at all. It should be entirely sufficient to wrap the native `SailRepositoryConfig` with a `LuceneSailConfig` and then using the additional properties which this sail makes accessible in SPARQL queries: http://docs.rdf4j.org/programming/#_full_text_indexing_with_the_lucene_sail